src/wrapped_radix_engine_toolkit/high_level/manifest_building/modify_manifest.rs

```diff
@@ -66,21 +66,20 @@ impl TransactionManifest {
     ///
     /// Also panics if the number of TransactionGuarantee's is larger than the number
     /// of instructions of `manifest` (does not make any sense).
-    pub(crate) fn modify_add_guarantees<I>(self, guarantees: I) -> Self
+    pub(crate) fn modify_add_guarantees<I>(
+        self,
+        guarantees: I,
+    ) -> Result<TransactionManifest>
```

Comment:
From what we've discussed on Slack, the only reason we add this `Result<>` is so that UniFFi converts it into a throwable method when translated into Swift. If that's the case, one could be tempted to remove the `Result<>` from this function as it is never used. Therefore, to avoid this issue coming back in the future, I'd recommend:
- adding a comment explaining why we need to use `Result<>`
- update line 88 so that instead of `panic!`, we return some `Err` and then catch such error on the host side. Would that work?

Reply:
Returning an error should work, though then we'd need to add that error to CommonError. If we did that, we wouldn't have to add a comment though, since the Result type would make sense.

Reply:
right, I meant one of the two options

Reply:
Got it. The second one seems to make sense either way. It is very strange that Uniffi converts the panic into a secret Err.

Reply:
Agree, we should replace the specific panic! in this function with a CommonError.

Reply:
Added IndexOutOfBounds to CommonError